src/client.ts

```diff
@@ -30,23 +34,28 @@ import {
   isQueryFailure,
   isQuerySuccess,
   type QueryFailure,
-  type QueryRequest,
-  type QueryRequestHeaders,
   type QuerySuccess,
   type QueryValue,
-  type ValueFormat,
 } from "./wire-protocol";
 
-interface RequiredClientConfig {
-  client_timeout_buffer_ms: number;
-  endpoint: URL;
-  format: ValueFormat;
-  http2_session_idle_ms: number;
-  http2_max_streams: number;
-  fetch_keepalive: boolean;
-  secret: string;
-  query_timeout_ms: number;
-}
+type RequiredQueryOptions = QueryOptions &
+  Required<Pick<QueryOptions, "format" | "query_timeout_ms">>;
```

Comment:
do we need this type? or is typescript smart enough to just infer this when you do the `{ ...clientConfiguration, ...queryOptions }` in `#query`?

Reply:
Yep! By trying to assign a type to `requestConfig` I made it too specific. Like you were getting at, `Client.clientConfiguration` already has these fields set to required, so typescript can figure that out if I just let it infer the type.